pylabnet/scripts/sweeper/scan_fit.py

```diff
@@ -0,0 +1,194 @@
+""" Class for fitting popup for scan_1d.py """
+from pylabnet.gui.pyqt.external_gui import Popup, InternalPopup
+from scipy.optimize import curve_fit # is not automatically installed?
+import numpy as np
+
```

Comment:
i have some minor concerns about how individual model-specific methods are used in `FitPopup`. This means that each time we need to add a new model, we need to extend this class substantially, in addition to drawing a new ui file. In principle, all the information can be contained in the definition of the model: a functional form, and designation of parameters to fit.

Do you think it would be possible to modify the code so that to create a new model, minimal code-writing is required? This would greatly improve the utility + useability of the fitting. Happy to discuss how this might be implemented